File: ExampleApp/app/src/main/java/com/motiv/example/GoApi.java
package com.motiv.example;

import androidx.fragment.app.*;
import com.google.gson.*;
import com.google.gson.annotations.*;
import com.google.gson.reflect.*;
import dagger.*;
import dagger.android.*;
import dagger.android.support.*;
import javax.inject.*;
import retrofit2.*;
import retrofit2.http.*;

public class GoApi {

    private static GoApiApi goApiApi;

    public GoApi(GoApiApi goApiApi) {
        this.goApiApi = goApiApi;
    }

    public void getUser(
            java.lang.String userId,
            final com.motiv.example.OnResponseListener<com.motiv.example.UserResponse>
                    onResponseListener) {

        goApiApi.getUser(userId)
                .enqueue(
                        new Callback<com.motiv.example.UserResponse>() {
                            @Override
                            public void onResponse(
                                    Call<com.motiv.example.UserResponse> call,
                                    retrofit2.Response<com.motiv.example.UserResponse> response) {

                                onResponseListener.onSuccess(response.body());
                            }

                            @Override
                            public void onFailure(
                                    Call<com.motiv.example.UserResponse> call, Throwable t) {
                                onResponseListener.onError(new Exception(t.getMessage()));
                            }
                        });
    }

    public void getUsersList(
            final com.motiv.example.OnResponseListener<com.motiv.example.UsersResponse>
                    onResponseListener) {
        goApiApi.getUsersList()
                .enqueue(
                        new Callback<com.motiv.example.UsersResponse>() {
                            @Override
                            public void onResponse(
                                    Call<com.motiv.example.UsersResponse> call,
                                    retrofit2.Response<com.motiv.example.UsersResponse> response) {

                                onResponseListener.onSuccess(response.body());
                            }

                            @Override
                            public void onFailure(
                                    Call<com.motiv.example.UsersResponse> call, Throwable t) {

                                onResponseListener.onError(new Exception(t.getMessage()));
                            }
                        });
    }

    public void getPostsList(
            final com.motiv.example.OnResponseListener<com.motiv.example.PostsListResponse>
                    onResponseListener) {
        goApiApi.getPostsList()
                .enqueue(
                        new Callback<com.motiv.example.PostsListResponse>() {
                            @Override
                            public void onResponse(
                                    Call<com.motiv.example.PostsListResponse> call,
                                    retrofit2.Response<com.motiv.example.PostsListResponse>
                                            response) {

                                onResponseListener.onSuccess(response.body());
                            }

                            @Override
                            public void onFailure(
                                    Call<com.motiv.example.PostsListResponse> call, Throwable t) {

                                onResponseListener.onError(new Exception(t.getMessage()));
                            }
                        });
    }

    public void getPhotos(
            final com.motiv.example.OnResponseListener<com.motiv.example.PhotosListResponse>
                    onResponseListener) {
        goApiApi.getPhotos()
                .enqueue(
                        new Callback<com.motiv.example.PhotosListResponse>() {
                            @Override
                            public void onResponse(
                                    Call<com.motiv.example.PhotosListResponse> call,
                                    retrofit2.Response<com.motiv.example.PhotosListResponse>
                                            response) {

                                onResponseListener.onSuccess(response.body());
                            }

                            @Override
                            public void onFailure(
                                    Call<com.motiv.example.PhotosListResponse> call, Throwable t) {

                                onResponseListener.onError(new Exception(t.getMessage()));
                            }
                        });
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/GoApi.kt
 
package com.motiv.example
import androidx.fragment.app.*
import com.google.gson.*
import com.google.gson.annotations.*
import com.google.gson.reflect.*
import dagger.*
import dagger.android.*
import dagger.android.support.*
import javax.inject.*
import retrofit2.*
import retrofit2.http.*

public class GoApi(val goApiApi: GoApiApi) {

    fun getUser(userId: String, onResponseListener: com.motiv.example.OnResponseListener<com.motiv.example.UserResponse>) {
        goApiApi!!.getUser(userId).enqueue(object : Callback<com.motiv.example.UserResponse> {
            override fun onFailure(call: Call<com.motiv.example.UserResponse>?, t: Throwable?) {
                onResponseListener.onError(Exception(t?.message))
            }

            override fun onResponse(call: Call<com.motiv.example.UserResponse>?, response: Response<com.motiv.example.UserResponse>?) {
                onResponseListener.onSuccess(response?.body() as com.motiv.example.UserResponse)
            }
        })
    } fun getUsersList(onResponseListener: com.motiv.example.OnResponseListener<com.motiv.example.UsersResponse>) {
        goApiApi!!.getUsersList().enqueue(object : Callback<com.motiv.example.UsersResponse> {
            override fun onFailure(call: Call<com.motiv.example.UsersResponse>?, t: Throwable?) {
                onResponseListener.onError(Exception(t?.message))
            }

            override fun onResponse(call: Call<com.motiv.example.UsersResponse>?, response: Response<com.motiv.example.UsersResponse>?) {
                onResponseListener.onSuccess(response?.body() as com.motiv.example.UsersResponse)
            }
        })
    } fun getPostsList(onResponseListener: com.motiv.example.OnResponseListener<com.motiv.example.PostsListResponse>) {
        goApiApi!!.getPostsList().enqueue(object : Callback<com.motiv.example.PostsListResponse> {
            override fun onFailure(call: Call<com.motiv.example.PostsListResponse>?, t: Throwable?) {
                onResponseListener.onError(Exception(t?.message))
            }

            override fun onResponse(call: Call<com.motiv.example.PostsListResponse>?, response: Response<com.motiv.example.PostsListResponse>?) {
                onResponseListener.onSuccess(response?.body() as com.motiv.example.PostsListResponse)
            }
        })
    } fun getPhotos(onResponseListener: com.motiv.example.OnResponseListener<com.motiv.example.PhotosListResponse>) {
        goApiApi!!.getPhotos().enqueue(object : Callback<com.motiv.example.PhotosListResponse> {
            override fun onFailure(call: Call<com.motiv.example.PhotosListResponse>?, t: Throwable?) {
                onResponseListener.onError(Exception(t?.message))
            }

            override fun onResponse(call: Call<com.motiv.example.PhotosListResponse>?, response: Response<com.motiv.example.PhotosListResponse>?) {
                onResponseListener.onSuccess(response?.body() as com.motiv.example.PhotosListResponse)
            }
        })
    }
}
